lib/Sema/CSDiagnostics.h

```diff
@@ -158,6 +160,12 @@ class FailureDiagnostic {
 
   Optional<SelectedOverload> getChoiceFor(Expr *) const;
 
+  /// If this is a failure for a function argument conversion, or a constraint
+  /// within an argument conversion, returns information about the application
+  /// of the argument to its parameter. If this failure is not for an argument
+  /// conversion, returns \c None.
+  Optional<FunctionArgApplyInfo> getFunctionArgApplyInfo() const;
```

Comment:
I think it would be great for extensibility if this method would take a `ConstraintLocator *` as a parameter.

Reply:
@xedin Hah, that's actually how I originally had it, but wasn't sure whether or not it was worth it :) I'll go ahead and add that.

Thanks for the review!